### ISSUE TITLE
CMake 3.10 requirement

### ISSUE THREAD

**Issue by ntrost57** [ROCmSoftwarePlatform/rocPRIM#3]
Hi, is there a special reason for the cmake 3.10 requirement? Ubuntu 16.04 hosts only version 3.5. Also, cmake 3.10 fails to find rocm package, whereas 3.5 works fine for me.

**Comment by jszuppe** [comment]
The main reason was that 3.10 was the newest version at the time we started developing rocPRIM. We didn't see any reason to use an old tool, when there is a new one, and installing new CMake is very easy. Additionally, we got some problems with CMake 3.5 and 3.8 (something worked for one, but not for the other) during rocRAND development.

To be honest, I don't know why it does not work for you. We have about 5-7 different machines with cmake 3.10 and ROCm + docker for CI, and we've never had a problem with finding ROCm package. Also failure in finding ROCm package should not cause build failure, because in such cases required rocm-cmake project is automatically downloaded.

**Comment by ntrost57** [comment]
This is my output using cmake 3.5.1
```
cmake ..
-- Found HIP: /opt/rocm (found version "1.5.18081") 
-- The CXX compiler identification is Clang 6.0.0
-- Check for working CXX compiler: /opt/rocm/bin/hcc
-- Check for working CXX compiler: /opt/rocm/bin/hcc -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Setting build type to 'Release' as none was specified.
-- Found Git: /usr/bin/git (found version "2.7.4") 
-- Found PkgConfig: /usr/bin/pkg-config (found version "0.29.1") 
-- Found GTest: /usr/local/lib/libgtest.a  
-- ******** Summary ********
-- General:
--   System                : Linux
--   HIP ROOT              : /opt/rocm
--   C++ compiler          : /opt/rocm/bin/hcc
--   C++ compiler version  : 6.0.0
--   CXX flags             : 
--   Build type            : Release
--   Install prefix        : /opt/rocm
-- 
--   BUILD_TEST            : ON
--   BUILD_BENCHMARK       : OFF
--   BUILD_EXAMPLE         : OFF
-- Configuring done
-- Generating done
-- Build files have been written to: ./build
```

and this using cmake 3.11.0
```
/opt/cmake-3.11.0/bin/cmake ..
-- Found HIP: /opt/rocm (found version "1.5.18081") 
-- The CXX compiler identification is Clang 6.0.0
-- Check for working CXX compiler: /opt/rocm/bin/hcc
-- Check for working CXX compiler: /opt/rocm/bin/hcc -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Setting build type to 'Release' as none was specified.
-- Found Git: /usr/bin/git (found version "2.7.4") 
-- Found PkgConfig: /usr/bin/pkg-config (found version "0.29.1") 
-- Found GTest: /usr/local/lib/libgtest.a  
CMake Error at cmake/Dependencies.cmake:138 (find_package):
  Could not find a package configuration file provided by "ROCM" with any of
  the following names:

    ROCMConfig.cmake
    rocm-config.cmake

  Add the installation prefix of "ROCM" to CMAKE_PREFIX_PATH or set
  "ROCM_DIR" to a directory containing one of the above files.  If "ROCM"
  provides a separate development package or SDK, be sure it has been
  installed.
Call Stack (most recent call first):
  CMakeLists.txt:63 (include)


-- Configuring incomplete, errors occurred!
See also "./build/CMakeFiles/CMakeOutput.log".
See also "./build/CMakeFiles/CMakeError.log".
```

**Comment by VincentSC** [comment]
Did you install the rocm-module for cmake 3.10? https://github.com/RadeonOpenCompute/rocm-cmake

**Comment by jszuppe** [comment]
I've just tested 3.11.1 on CUDA machine (where rocm-cmake is always downloaded) and it works ok. Checking ROCm platform now.

**Comment by jszuppe** [comment]
It works on ROCm platform and a docker-based CI too. Do you have `rocm-cmake` package installed? Is there `rocm-cmake-master` folder in your `build` folder after you run cmake script?

**Comment by jszuppe** [comment]
It's also possible that somehow download command or zip extraction fail for you, but I assume it would fail on 3.5 too.

**Comment by ntrost57** [comment]
I am going to have look at that. Seems download worked, but it did not unzip the file, build directory only holds the .zip file.
cmake 3.5 works fine (also downloading and unzip).

**Comment by jszuppe** [comment]
Hmm.. I really don't know why unziping works on 3.5 but not on 3.11 for you.

**Comment by ntrost57** [comment]
If there is no show stopper, I think we should move back to cmake 3.5, also because it is the standard available version with ubuntu 16.04.

**Comment by VincentSC** [comment]
There were problems with handling CUDA in cmake 3.5.1, but they're fixed for the final release (ETA halfway June). When you're only working with AMD devices you can safely use cmake 3.5.1.
